specs/eventing/data-plane.md

```diff
@@ -1,125 +1,184 @@
+  take a stance here.
 
-If an HTTP request's URL does not correspond to an existing endpoint, then the
-Sink MUST respond with `404 Not Found`.
+- The CloudEvents HTTP bindings provide a relatively simple and efficient
+  network protocol which can easily be supported in a wide variety of
+  programming languages leveraging existing library investments in HTTP. The
+  CloudEvents project has already written these libraries for many popular
+  languages.
 
-Every non-Callable Sink MUST respond with `202 Accepted` if the request is
-accepted.
+- Knative Eventing assumes a sender-driven (push) event delivery system. That
+  is, each recipient is actively responsible for an event until it is handled
+  (or affirmatively delivered to all following recipients).
 
-If Sink is Callable it MAY respond with `200 OK` and a single event in the HTTP
-response. A returned event is not required to be related to the received event.
-The Callable should return a successful response if the event was processed
-successfully. If there is no event to send back then Callable Sink MUST respond
-with 2xx HTTP and with empty body.
+- Knative Eventing aims to make [event sources](./overview.md#event-source) and
+  event-processing software easier to write; as such, it imposes higher
+  standards on system components like [brokers](./overview.md#broker) and
+  [channels](./overview.md#channel) than on edge components.
 
-If a Sink receives a request and is unable to parse a valid CloudEvent, then it
-MUST respond with `400 Bad Request`.
+This contract defines a mechanism for a single event sender to reliably deliver
+a single event to a single recipient. Building from this primitive, chains of
+reliable event delivery and event-driven applications can be built.
 
-### Content Modes Supported
+## Background
 
-A Sink MUST support `Binary Content Mode` and `Structured Content Mode` as
-described in
-[HTTP Message Mapping section of HTTP Protocol Binding for CloudEvents](https://github.com/cloudevents/spec/blob/master/http-protocol-binding.md#3-http-message-mapping)
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD",
+"SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be
+interpreted as described in
+[RFC2119](https://datatracker.ietf.org/doc/html/rfc2119).
 
-A Sink MAY support `Batched Content Mode` but that mode is not used in Knative
-Eventing currently (that may change in future).
+When not specified in this document, the
+[CloudEvents HTTP bindings, version 1.0](https://github.com/cloudevents/spec/blob/v1.0.1/http-protocol-binding.md)
+and [HTTP 1.1 protocol](https://tools.ietf.org/html/rfc7230) standards MUST be
+followed (with the CloudEvents bindings taking precedence in the case of
+conflict).
 
-### Retries
+The current version of this document does not describe protocol negotiation or
+any delivery mechanism other than HTTP 1.1. Future versions might define
```

Comment:
There's no normative words here, but someone could interpret this as HTTP/1.1 MUST be used but I don't think that's necessary true, is it?  Can someone only support HTTP/2 and still be compliant?

Reply:
I think you would need to implement HTTP/1.1, but you COULD offer an upgrade to HTTP/2. (HTTP/1.1 and HTTP/2 are not compatible, but you can upgrade from the former to the latter.)